### ISSUE TITLE
Eviction incorrectly forbids evicting an unready pod that wouldn't affect the PDB

### ISSUE THREAD

**Issue by tnozicka** [kubernetes/kubernetes#103970]
#### What happened:
Our operator had a StatefulSet with 3 pods, and a PDB with `maxUnavailable: 1`. First 2 pods were ready, the 3rd one was not ready (and will never be ready). The operator tried to evict the 3rd (unready) pod, but the API call was repeatedly forbidden with `Cannot evict pod as it would violate the pod's disruption budget`. Yet, that isn't true - the PDB accounts only for ready pods (the first 2) so the 3rd one that isn't ready should be evictable without an effect on the PDB.

#### What you expected to happen:
Unready pods should be freely evictable as they are not accounted for in the PDB (currentHealthy).

#### How to reproduce it (as minimally and precisely as possible):
1. Create 2 ready pods and 1 unready
2. Create a PDB with `maxUnavailable: 1`
3. Try to evict the unready pod

#### Anything else we need to know?:
For more context this originates from https://github.com/scylladb/scylla-operator/issues/730#issuecomment-888030132
The full yamls can be seen at https://cloudius-jenkins-test.s3.amazonaws.com/801e256a-2bca-4c72-be78-40711c7838c9/20210724_185703/kubernetes-801e256a.tar.gz but the relevant parts are the following:
```
apiVersion: v1
kind: Pod
metadata:
  name: sct-cluster-us-east1-b-us-east1-2
  namespace: scylla
status:
  - lastProbeTime: null
    lastTransitionTime: "2021-07-24T18:14:09Z"
    message: 'containers with unready status: [scylla scylla-manager-agent]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
```
```
apiVersion: policy/v1beta1
kind: PodDisruptionBudget
metadata:
  name: sct-cluster
  namespace: scylla
spec:
  maxUnavailable: 1
status:
  currentHealthy: 2
  desiredHealthy: 2
  disruptionsAllowed: 0
  expectedPods: 3
  observedGeneration: 1
```
```
E0724 18:39:12.276969       1 scyllacluster/controller.go:189] syncing key 'scylla/sct-cluster' failed: Cannot evict pod as it would violate the pod's disruption budget.
I0724 18:39:12.276985       1 record/event.go:291] "Event occurred" object="scylla/sct-cluster-us-east1-b-us-east1-2" kind="PersistentVolumeClaim" apiVersion="v1" type="Warning" reason="DeletePersistentVolumeClaimFailed" message="Failed to delete PersistentVolumeClaim scylla/sct-cluster-us-east1-b-us-east1-2: Cannot evict pod as it would violate the pod's disruption budget."
```

#### Environment:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-06-16T12:59:11Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.12-gke.1100", GitCommit:"6d7f996b31f221c09b59426b03567e62061c7f46", GitTreeState:"clean", BuildDate:"2021-07-02T09:24:42Z", GoVersion:"go1.15.13b5", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: GKE

/sig apps
/kind bug
/priority important-soon
/assign

@kubernetes/sig-apps-bugs

**Comment by tnozicka** [comment]
looking at the code, this looks fixed since 1.20, but never made it to 1.19

**Comment by kerthcet** [comment]
I do not support the idea yet.🤧  PDB is help to limit the number of Pods  down simultaneously by voluntary disruptions, it is focus on ha not scheduling, separation of concerns i think. And can I know about the reason what causes the pod not-ready? @tnozicka
